Improve analysis of Heracles.ts

priyanshunayan opened the pull request:
Following are the changes included in this PR:
1. Included Usage instruction of Heracles.ts. Now one can install and use Heracles.ts on local machine in either browser or in node.js environment. Before it was difficult to figure out the environment requirements and dependencies required along side Heracles in order to use it. 
2. More clear explanation of classes used in Heracles.ts. Earlier it was without any context which made it difficult to understand. Now one can run and see what each class is doing. This doc serves as the starting point.
3. Removed Outdated classes like ResourceEnrichmentProvider.

xadahiya commented on pages/heracles_explained.md:
```diff
@@ -3,101 +3,203 @@ layout: page
-Actually, there are many interfaces which have track on the objects as well as its properties as we have disussed above and with the implementation of the interface (called Class above) means accessing the members or properties of interface for whom  client can get a lot of knowledge about the starting url or server.
-
-Load on server and speed can be limitation for the client. But in [python-hydra-agent](https://github.com/HTTP-APIs/python-hydra-agent)  we are dicsussing mainly about how to reduce the load on server and fast querying from the client. For this type of problem, we are using Redis to reduce load of server and we can implement indexing (specially secondary/faceted indexing) of objects and their pro ...
+To better understand things, let’s do it by using Heracles.ts. We can use Heracles.ts in the browser as well as in the node environment.
```
We should try to keep the docs in the third person as much as possible. Use something like "Heracles.ts can be installed ..."

xadahiya replied:
The same goes for the whole page

xadahiya commented on pages/heracles_explained.md:
````diff
@@ -3,101 +3,203 @@ layout: page
+.withStrictLinks()
+.with(fetch.bind(global || window));
+}
+```
````
Since Heracles.ts is not our tool, I don't think we need an installation guide or anything. Let me explain why this page is there, as you already know Heracles.ts is the agent built by hydraCG in ts, this page is supposed to be a brief explanation how Heracles.ts is structured, built, what features it has, etc so that developers can take design clues from Heracles.ts architecture in order to improve our own agent. It would be awesome if you could structure this page keeping this goal in mind.

priyanshunayan replied:
Should I remove both browser and node.js installation instructions? The reason why i put it there was because running and seeing things solidified my understanding. So, i thought it can help others too especially people who are just starting out.

xadahiya replied:
I guess we can keep it. Add this to the bottom of the page or somewhere nit on the top

priyanshunayan commented:
@xadahiya as asked I have changed the narration to third person, add index for Heracles.ts and explained them in brief, Shifted installation guides to bottom. Also removed node.js installation guide for now as they are going to release a new version soon that will support node.js out of the box.
